x-pack/plugins/observability_solution/observability_ai_assistant/server/service/util/get_category_query.ts

```diff
@@ -27,7 +27,7 @@ export function getCategoryQuery({ categories }: { categories?: string[] }) {
           noCategoryFilter,
           {
             terms: {
-              'labels.category': categories,
+              'labels.category.keyword': categories,
```

Comment:
just for my understanding, does this mean it was not working before? Looking at this, we probably shouldn't use labels for this, but I'm not sure if we have a migration process right now

Reply:
> does this mean it was not working before?

It is possible to use a `terms` query on a `text` field, you're just likely to not find a match because the `text` field has been analyzed so the value might have changed, but since our values here are just single words I doubt the analyzer actually changed anything about them.

> Looking at this, we probably shouldn't use labels for this, but I'm not sure if we have a migration process right now

Can you expand on these two points (why not use labels, migration process)?

Reply:
>It is possible to use a terms query on a text field, you're just likely to not find a match because the text field has been analyzed so the value might have changed, but since our values here are just single words I doubt the analyzer actually changed anything about them.

that makes sense 

> Can you expand on these two points (why not use labels, migration process)?

Labels are mapped unpredictably (first-come first-serve etc). But we know what we want to store here and we depend on it functionality. `labels` are better off as a tagging mechanism for the user. 

If we want to use a known field (e.g. simply `category`), we would have to migrate existing documents to use `category` instead of `labels.category`. I don't think we have that yet.

Reply:
I filed https://github.com/elastic/kibana/issues/180446 to address this part